webapp/loggedin/modules/data/dataVis/dataQuery/components/tableColumnEdit.js

```diff
@@ -28,26 +31,33 @@ const TableColumnEdit = (props) => {
       [Node.keys.parentUuid]: parentUuid,
     }
 
+    const recordValidation = Record.getValidation(record)
+    const validation = RecordValidation.getNodeValidation(node)(recordValidation)
+
     return (
-      <NodeDefTableCellBody
-        surveyInfo={surveyInfo}
-        surveyCycleKey={surveyCycleKey}
-        nodeDef={nodeDef}
-        parentNode={parentNode}
-        nodes={[node]}
-        entry
-        entryDataQuery
-        edit={false}
-        renderType={NodeDefLayout.renderType.tableBody}
-        canEditRecord={canEditRecord}
-        updateNode={(...args) => dispatch(updateNode(...args))}
-        removeNode={(...args) => dispatch(removeNode(...args))}
-        createNodePlaceholder={(...args) => dispatch(createNodePlaceholder(...args))}
-        windowed={false}
-      />
+      <div>
+        {!Validation.isValid(validation) && (
+          <ErrorBadge validation={validation} showLabel={false} className="error-badge-inverse" />
```

Comment:
does this add the warning icon? this might look ugly, only add background color red (add a parameter to errorbadge if there isn't showIcon

Reply:
why adding an external div? can it wrapped inside error badge?

Reply:
ok